GoogleForCreators/web-stories-wp
# Editor: Hidden Padding for Text Fill & Highlight Elements

## Summary
Adds a padding to text elements with bg fill and highlight that appears on the element, but is not visible in the inspector. The hidden padding is currently set to 6x2 in editor coordinates.

## Relevant Technical Choices
Applied the extra padding to the elements padding in the story data so that we didn't have to alter our positioning system at all, then added a flag that hidden padding is present on the element data so we can subtract it out in the padding input.

Wanted to maintain backwards compatibility, so if users have an existing story before this change with `0` padding on bg fill text that's pixel perfect, this doesn't alter it, but if they reapply fill or highlight to the text, the updated hidden padding will be present.

## To-do
NA

## User-facing changes
Text with bg fill or highlight should now have a little extra spacing around the text to be visually pleasant to user. Other than that there should be no visible changes to the inspector panel behavior.

## Testing Instructions
-> Go to story editor
-> Add text element
-> Apply BG Fill to text and see that the bounding box grows a little bit, but still has a value of `0` in the padding input
-> Apply BG highlight from BG Fill an see that element retains the same size and `0` for input padding.
-> Mess with different combinations of selected elements (locked, unlocked, same padding, different padding) and see that the padding input has not changed (There's a lot of combinations of scenarios you can create here, but don't worry if you don't think of all of them because the unit tests are pretty thorough going through these scenarios)

---



Fixes #4972 
See #4980
